### PR TITLE
Adding PUBLISHER_FROM_EMAIL for sending emails.

### PR DESCRIPTION
**Pull request by awais786**
## Configuration Pull Request

Make sure that the following steps are done before merging
- [x] @devops team member has commented with :+1:
- [x] are you adding any new default values that need to be overriden when this goes live?  
  - [x] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
  - [x] Add an entry to the CHANGELOG.

ECOM-6141

### CODE REVIEW

**Comment by awais786** [comment]
This is the email address `publisher-notification@edx.org`

**Comment by awais786** [comment]
Ticket for DEVOPS
https://openedx.atlassian.net/browse/DEVOPS-4975

**Comment by awais786** [comment]
@edx/ecommerce Please review this PR.

**Review by bderusha — APPROVED** [review]

File: playbooks/roles/discovery/defaults/main.yml [hunk 1]
```diff
@@ -119,6 +119,8 @@ DISCOVERY_EMAIL_USE_TLS: False
 DISCOVERY_EMAIL_HOST_USER: ''
 DISCOVERY_EMAIL_HOST_PASSWORD: ''
 
+DISCOVERY_PUBLISHER_FROM_EMAIL: 'publisher@example.com'
```
Comment (clintonb) [inline playbooks/roles/discovery/defaults/main.yml]: Can we set this to null so that no emails are sent if the value is not properly defined?

**Review by clintonb — CHANGES_REQUESTED** [review]
We should avoid fake data that seems real, so it's obvious there are errors and the system fails in a manner that we can find those errors during smoke testing.

**Review by fredsmith — APPROVED** [review]

**Comment by awais786** [comment]
@fredsmith should I merge this PR or wait for the DEVOPS ticket to get complete ?

**Comment by awais786** [comment]
@clintonb Please review this PR again.

**Review by clintonb — APPROVED** [review]

**Pull request merged**